x-pack/plugins/security/server/routes/views/access_notice.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RouteDefinitionParams } from '..';
+
+/**
+ * Defines routes required for the Access Notice view.
+ */
+export function defineAccessNoticeRoutes({
+  authc,
+  httpResources,
+  config,
+  router,
+  logger,
+}: RouteDefinitionParams) {
+  httpResources.register(
+    { path: '/security/access_notice', validate: false },
+    async (context, request, response) => response.renderCoreApp()
+  );
+
+  router.get(
+    { path: '/internal/security/access_notice/state', validate: false },
+    async (context, request, response) => {
+      // It's not guaranteed that we'll have session for the authenticated user (e.g. when user is
+      // authenticated with the help of HTTP authentication), that means we should safely check if
+      // we have it and can get a corresponding configuration.
+      try {
+        const session = await authc.getSessionInfo(request);
+        const accessNotice =
+          (session &&
+            (config.authc.providers as Record<string, any>)[session.provider.type]?.[
```

Comment:
__note:__ extracting config from `config.authc.providers` is a bit ugly since we don't know what providers are configured and there is no "type link" between session provider field and this config (and may not be since session can rely on old config). If we see this pattern more and more, we'll create a helper method instead.